### DISCUSSION TITLE
variant analysis and doubt

### DISCUSSION THREAD

**Discussion by mugikumar** [ORTools4MTEX/ORTools#17] [Q&A] [answered]
Hi, frank,
          Thank you for sharing the code of variant analysis in lath martensite. I doubt the variant analysis in a single grain. I want to analyze 24 variants in a single grain. it's showing the error of the inverse pole figure for individual grains

**Comment by frankNiessen** [comment, marked as answer]
Hi mugikumar, 

i am happy to help, but I think you need to be more specific of what you are computing, what your exptected output is, and what the actual output is for me to be able to support you. 

Best wishes
Frank

**Reply by mugikumar** [reply]
hi frank, thanks for your reply. when I was run the variant analysis section, it has shown an error in the packet id calculation. I attached the screenshot below
        
![Untitled](https://user-images.githubusercontent.com/109027461/188150751-b823e335-c7f0-49a5-b00d-4944b6320543.jpg)

**Reply by frankNiessen** [reply]
Hi again,

your error message states *out of memory*. How large is your map and what kind of computer are you running on?
This is likely a general MATLAB issue and not related to our program, have a look [here](https://se.mathworks.com/help/matlab/matlab_prog/resolving-out-of-memory-errors.html). 

Best wishes
Frank

**Reply by mugikumar** [reply]
Thanks once again, Frank. I resolved those out-of-memory errors because my map has a more number of orientations, nearly 10 million. I was getting the error on running variant information for a PAG and didn't get the inverse pole figure of individual grains. I have uploaded the image. Can you please help me to get out of these errors? 

![error](https://user-images.githubusercontent.com/109027461/188263360-253ad9f8-82e2-4e27-b4c6-da8ab5c96e8c.jpg)
![error2](https://user-images.githubusercontent.com/109027461/188263373-658316b2-f132-460f-be5f-732c5ac2d8c4.jpg)
![error3](https://user-images.githubusercontent.com/109027461/188263377-9a26fda4-6a93-42a8-a199-a7937e75467e.jpg)

**Reply by frankNiessen** [reply]
Dear mugikumar, 

which version of ORTools are you running? I just checked my code and it works. Specifically, the code you show does not match the latest code anymore, I recall that this has been an error that has been ironed out a good while ago. The current code looks like this:

```
hold all
plot(pGrain.boundary,...
    'lineWidth',1,'lineColor',[0.5 0.5 0.5]);
[~,mP] = plot(cGrains.boundary,...
    'lineWidth',1,'lineColor',[0 0 0]);
hold off
```

Maybe you could install the [latest release](https://github.com/ORTools4MTEX/ORTools/releases/tag/2.0.0) and let me know if it works for you?

Best wishes
Frank

**Reply by mugikumar** [reply]
Dear Frank,
                 I am sorry to disturb you again; I still get errors while using the new ORTools 2.0. Every time, I got the error of IPF of individual grains and the error in color names. I uploaded the error image down. Could you please help me out?
![error in pdf](https://user-images.githubusercontent.com/109027461/188610219-34289a70-9d13-4ab5-89ce-76073b982d82.jpg)

**Reply by frankNiessen** [reply]
Hi again, 

is it only in *plotStack* that you have this problem? Can you plot other IPF maps? It seems to me that this is an error that has its roots in MTEX. Have you tried running some basic MTEX code?

Best wishes
Frank

**Reply by mugikumar** [reply]
Hi Frank,
                I was getting the problem while running on _plotstack_. I didn't get any trouble running on the IPF figure of the whole EBSD map and got all images when I was running basic MTEX.  I have taken all photos of the PAG reconstruction. I think so might have a problem when was imported the OSC file. Can you suggest to me how to import osc file

**Reply by frankNiessen** [reply]
I do not get that, if you can plot all other maps your file import should be ok?
You can use the [import wizard](https://mtex-toolbox.github.io/EBSDImport.html) and generate a script to import your data.

**Reply by mugikumar** [reply]
Hi Frank, thanks for your response. After I deleted the error code of p(EBSD.orientation), it's working fine. I got the individual grains histogram of the variant id plot. So can you please look into that code to change it to some other way?
This coding plot has some problems. If am wrong please leave it.
%% Plot the parent orientation PDF
drawnow;
figH = gobjects(1);
figH = figure('WindowStyle','docked');
set(get(handle(figH),'javaframe'),'GroupName',dockGroupName);
drawnow;
if check_option(varargin,'grains')
    plotPDF(pGrain.meanOrientation,...
        hParent,...
        'equal','antipodal',...
        'MarkerSize',10,'MarkerFaceColor',[1 1 1],...
        'lineWidth',1,'MarkerEdgeColor',job.csParent.color);
else
    plotPDF(pEBSD.orientations,...
        hParent,...
        'equal','antipodal',...
        'MarkerSize',10,'MarkerFaceColor',[1 1 1],...
        'lineWidth',1,'MarkerEdgeColor',job.csParent.color);
end
hold off
set(figH,'Name','PDF: Parent grain orientation','NumberTitle','on');
drawnow;

**Reply by frankNiessen** [reply]
Hi mugikumar, 

I am glad to hear that it is working for you now. I still have problems reproducing your error even when trying several different datasets. I therefore do not know exactly how to change the code. If there are more people reporting the problem we will have to look into it in more detail.

Best wishes
Frank

**Reply by mugikumar** [reply]
Thanks frank for your support

**Comment by mugikumar** [comment]
Hi Frank,
                     I have to plot the histogram for overall variants and
packet id calculation. could you please share the code?

With regards
mugikumar.

**Comment by frankNiessen** [comment]
Hi, 

I think the discussion [here](https://github.com/ORTools4MTEX/ORTools/discussions/14) should help you with your problem.

BW
Frank

**Comment by mugikumar** [comment]
Hi Frank
            Thanks for Your reply. I was getting the error of making band
contrast when I was using the code of plot(ebsd,ebsd.bc). can you please
help me to get out of error?

with regards
Mugi kumar

**Reply by frankNiessen** [reply]
Yes, in *.osc files, which are associated with the EDAX EBSD systems, there is no property bandcontrast (bc). You should check for other options to plot, I think you should have image quality instead (iq)? Thank you for pointing this out, we will have to make our code general to support all manufacturers of EBSD systems.

**Comment by mugikumar** [comment]
Hi Frank
            Thanks for your reply. I have to know is possible to get the
band contrast or image quality map from Mtex. Is there any coding for
interaction energy of which variants are formed first?

**Comment by mugikumar** [comment]
Sorry, I forget to include the file in ang format from mtex

**Comment by frankNiessen** [comment]
A quick Google search of **MTEX band contrast** should give you the answer. For ang files the equivalent property should be *iq*.

There is no direct function for this in Mtex, but you can use my code as a starting point:

[Click](https://github.com/frankNiessen/martensiteWorkCalculator)
